src/services/connections.service.ts

```diff
@@ -35,4 +35,35 @@ export class ConnectionsService {
 
 		return { data: connections, error: undefined };
 	}
+	static async get(connectionId: string): Promise<ServiceResponse<Connection>> {
+		let connectionResponse;
+		try {
+			const { connection } = await connectionsClient.get({ connectionId });
+			if (!connection || Object.keys(connection).length === 0) {
+				const errorMessage = translate().t("errors.connectionFetchFailed", { id: connectionId });
+				LoggerService.error(namespaces.deploymentsService, errorMessage);
+				return { data: undefined, error: errorMessage };
+			}
+
+			connectionResponse = convertConnectionProtoToModel(connection);
+		} catch (error) {
+			const errorMessage = translate().t("errors.connectionFetchFailed", { id: connectionId });
+			LoggerService.error(namespaces.deploymentsService, errorMessage);
+			return { data: undefined, error: errorMessage };
+		}
+		const { data: integrations } = await IntegrationsService.list();
```

Comment:
You always fetch all the integrations from the backend just for the integration name?

If so, please cache a map of integrationID to name somewhere and fetch it only once